state-chain/pallets/cf-broadcast/src/tests.rs

```diff
@@ -458,22 +388,16 @@ fn test_transmission_request_expiry() {
 		// Simulate the expiry hook for the expected expiry block.
 		let expected_expiry_block = current_block + BROADCAST_EXPIRY_BLOCKS;
 		Broadcaster::on_initialize(expected_expiry_block);
+		Broadcaster::on_idle(expected_expiry_block, 1_000_000_000_000_000u64.into());
```

Comment:
Could we use the test utilitie instead of this, like `then_execute_at_block` for example, this way we can ensure the hooks are executed in the correct order - more similar to how the chain actually runs, and we don't test cases that might not actually be possible

Reply:
Refactored this out. 
Also changed to other on_idle and on_initialize when it's possible (or when it makes sense)